File: Reqnroll/Chapter17/17.4-ExternalPersonaPattern/Before/WIMP.Specs/StepDefinitions/AuthenticationStepDefinitions.cs
using Reqnroll;

using WIMP.Specs.Drivers;
using WIMP.Specs.Support;

namespace WIMP.Specs.StepDefinitions;

[Binding]
public class AuthenticationStepDefinitions(AuthenticationApiDriver authApiDriver)
{
    [Given("the restaurant owner is authenticated")]
    public async Task GivenTheRestaurantOwnerIsAuthenticated()
    {
        await authApiDriver
            .Login(DomainDefaults.RestaurantOwner, DomainDefaults.Password)
            .Execute();
    }
}

File: CucumberJVM/Chapter16/16.4-ApprovedOutcomePattern/Before/src/test/java/com/wimp/app/specs/stepdefinitions/AuthenticationStepDefinitions.java
package com.wimp.app.specs.stepdefinitions;

import com.wimp.app.specs.drivers.AuthenticationApiDriver;
import com.wimp.app.specs.support.DomainDefaults;
import io.cucumber.java.en.*;

public class AuthenticationStepDefinitions {
    private final AuthenticationApiDriver authenticationApiDriver;

    public AuthenticationStepDefinitions(AuthenticationApiDriver authenticationApiDriver) {
        this.authenticationApiDriver = authenticationApiDriver;
    }

    @Given("the restaurant owner is authenticated")
    public void theRestaurantOwnerIsAuthenticated() throws Exception {
        authenticationApiDriver.login(DomainDefaults.RESTAURANT_OWNER, DomainDefaults.PASSWORD).execute();
    }
}
